File: 100-Doors/100_Doors.c
#include <stdio.h>
#include <stdbool.h>

/**
 * toggleDoors - Determines the state of the doors after 100 passes.
 * 
 * Return: Prints the door numbers that are open after the last pass.
 */
void toggleDoors() 
{
    bool doors[100] = {false};
    int i, j;

    for (i = 1; i <= 100; i++) 
    {
        for (j = i - 1; j < 100; j += i) 
	{
            doors[j] = !doors[j];
        }
    }

    printf("Open doors: ");
    for (int i = 0; i < 100; i++) 
    {
        if (doors[i]) 
	{
            printf("%d ", i + 1);
        }
    }
    printf("\n");
}

int main() 
{
    toggleDoors();
    return 0;
}

File: 100-Doors/100_doors.py
#!/usr/bin/python3
"""The module for 100 doors"""


def toggleDoors():
    """
    Determines the state of 100 doors after 100 passes.
    Returns a list of door numbers that remain open.
    """
    doors = [False] * 100
    for j in range(1, 101):
        for i in range(j - 1, 100, j):
            doors[i] = not doors[i]
    # Collect door numbers that are open (True)
    open_doors = [i + 1 for i in range(100) if doors[i]]
    return open_doors


open_doors = toggleDoors()
print(f"Open doors: {open_doors}")
